typescript-getting-started/functions/tsconfig.json

```diff
@@ -0,0 +1,17 @@
+{
+  "compilerOptions": {
+    "lib": ["es6", "es2015.promise"],
+    "module": "commonjs",
+    "noImplicitAny": false,
+    "outDir": "",
+    "sourceMap": true,
+    "target": "es5",
+    "typeRoots": [
+      "node_modules/@types"
+    ]
+  },
+  "include": [
```

Comment:
CC @mbleigh. Have you ever settled on a tsconfig file that you love? I forget whether this just dumps all files into the top-level directory or embeds everything as "$OUTDIR/[src|spec]/**/*.ts". I think our big options are:

1. Compile everything in-place so a .ts, .js, .js.map file exists at the same level. Pro: easy to set up, Con: hard to get clean builds, not all tools support file globs (e.g. gitignore can't specifically ignore *.js in nested directories of a single directory, so this breaks hybrid codebases)
2. Compile src & spec into a lib directory. Pros: easy. Cons: lib/src/*.js is confusing as heck
3. Use a tsconfig.json and a tsconfig.prod.json that inherits it. `lib` only contains compiled versions of `src` while `test` contains compiled versions of `src` and `spec`. Pros: production code doesn't have tests. Cons: more complex. Breaks anyone who assumes `tsc && firebase deploy` is the right command.

Reply:
yeah, I don't love dumping all the js files into the top level, but didn't know how to specify where `index.js` is if it is not at the root of the `functions` directory.  

The way this is set up now, all the `.ts` files are in `src` sub-directory, so at least that is something.